sdk/python/tests/integration/feature_repos/universal/data_sources/file.py

```diff
@@ -59,7 +59,17 @@ def create_data_source(
             field_mapping=field_mapping or {"ts_1": "ts"},
         )
 
-    def create_saved_dataset_destination(self) -> SavedDatasetFileStorage:
+    def create_saved_dataset_destination(
+        self, data_source: Optional[DataSource] = None
+    ) -> SavedDatasetFileStorage:
+        if data_source:
+            assert isinstance(data_source, FileSource)
+            return SavedDatasetFileStorage(
+                path=data_source.path,
+                file_format=ParquetFormat(),
+                s3_endpoint_override=None,
+            )
```

Comment:
Is this the main reason we're changing the interface for `create_saved_dataset_destination`? Can you explain what purpose the passed in data source serves?

Reply:
yeah so I want to copy an existing destination - however, each saved dataset object has a unique way of specifying destination (e.g. path vs. table vs. schema, etc.) so there's no super clean way to modify the `create_saved_dataset_destination` interface to allow for specifying a destination

the easiest thing to do is to just specify a data source, whose destination can be copied to a saved dataset object

Reply:
tbh i really dislike the idea of passing in a `source` as the `destination` :/ What else do you think we can pass in instead?

Reply:
switched to a different method, lmk what you think @achals